### PR TITLE
feat: update branding & button component

### PR DESCRIPTION
**Pull request by feralislatr**
**branding**

- add brand colors
- add brand fonts & begin implementation
- remove helvetica fonts
- add windi shortcut for button style
- update branding for button component -> https://github.com/BSA-US/dual-power-app/issues/74 
- add stories for button variants
- begin dark mode implementation

**tooling**

- update storybook config to allow custom fonts
- update gitignore

### CODE REVIEW

**Comment by feralislatr** [comment]
btw, my email is masked in my commits so that may be why the commit is showing as unverified.

**Comment by feralislatr** [comment]
trying to better organize the styles here.
i'm imagining that there should be extensions to existing classes or new shortcuts for the typography.

this is the info from the styleguide (modified to fit neater into rem sizing) coupled with applicable prose classes:
### large (n/a)
**font:** 36px / 2.25rem
**leading (line-height):** 115% -> 1.15rem 

### med (prose-2xl)
**font:** 24px / 1.5rem
**leading (line-height):** 125%: 1.25rem

### small (prose)
**font:** 16px / 1rem
**leading (line-height):** 115% -> 1.15rem


should text class names align with windi's prose classes or follow the hierarchy in our styleguide?

we have 3 levels of hierarchy in the styleguide--small (~16px), medium (24px), and large (36px).
windi's default prose classes align with a few, but they're a little different with `prose` being 16px and `prose-2xl` being 24px. as a new windi user, i'm not sure which would make the most sense to developers familiar with it.

**Comment by replygirl** [comment]
@feralislatr customizing `prose` is the move. `prose` lines up with `font-size` so all we need here is a `prose-4xl`

**Comment by replygirl** [comment]
got the lint action running again. looks like your editor isnt following the eslint config -- running `yarn lint --fix` might get that to pass

File: windi.config.ts [hunk 1]
```diff
@@ -75,13 +77,48 @@ export default defineConfig({
...
+        DARK: {
+          css: {
+            color: 'text-light-gray',
+          },
+        },
         DEFAULT: {
           css: {
-            color: colors.black,
-            lineHeight: 1.25,
+            color: 'text-black',
```
[the remark this answers is not in the document]
Comment (feralislatr) [reply windi.config.ts]: guess it wasn't. moved the colors out into a separate object so they can be referenced around the theme

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "@tsconfig/next/tsconfig.json",
   "compilerOptions": {
+    "jsx": "preserve",
```
[the remark this answers is not in the document]
Comment (feralislatr) [reply tsconfig.json]: my editor was screaming about something & made me add it, but it seems fine without it now

**Review by replygirl — APPROVED** [review]

**Pull request merged**